Update Calypso Colors

porada opened the pull request:
This branch updates all Calypso components and all color schemes with the [recently refreshed](https://color-studio.blog) palette colors.

## Notable changes

### Color schemes

Color scheme properties have been refactored:

* Color scheme stylesheets have been split into separate partials for better readability.
* All themable color properties are now prefixed with `--color-` while generic colors are directly outsourced to Color Studio properties (which are prefixed with `--studio-` and can’t be overwritten).
* Most color references in the code base have been outsourced to theme properties.
* Nightfall’s accent color has been changed from Purple to Blue due to WooCommerce folks’ wish to limit WordPress.com’s use of purple.
* All color combinations should be 100% accessible now.

### Components

Some components had to receive dedicated but non-breaking color treatments or refinements for better overall consistency:

* Badge
* Banner
* Color Scheme Picker
* Credit Card
* Jetpack Logo
* Push Notification Illustration
* Masterbar
* Notice
* Sidebar

### Other

* All the remaining color variables defined in SCSS have been phased out completely in favor of CSS custom properties.
* Updated SVG files for all plan icons and color scheme thumbnails. _No other SVG files have been touched._
* Removed unused code bits (where identified).

## Testing instructions

* Ensure color scheme switching works as intended.
* Ensure the components listed above are displayed correctly with all color schemes and under all supported browsers.
* Ensure this update doesn’t introduce any visual regressions.

porada committed:
Split color schemes into separate partials

porada committed:
Systematize custom colors

porada committed:
Refactor component-specific color property names

porada committed:
Refactor theme colors

porada committed:
Upgrade Color Studio

porada committed:
Tweak masterbar colors

porada committed:
Tweak individual themes

porada committed:
Colorize components

porada committed:
Colorize notice bars

porada committed:
Colorize plan banners and icons

porada committed:
Colorize scheme picker

porada committed:
Phase out SCSS color variables

porada committed:
Ensure Jetpack button colors are accessible

porada commented:
Regarding this bit:

> Updated SVG files for all plan icons and color scheme thumbnails. _No other SVG files have been touched._

Given these updates’ impact on our app accessibility, I think this branch shouldn’t have to wait for all illustrations to be updated.

Maybe we could outsource the SVG color updates to one of our creative technologists? I imagine this isn’t the last time we’ll have to perform those.

drw158 commented on apps/notifications/src/panel/boot/stylesheets/main.scss:
```diff
@@ -3,22 +3,22 @@
 
 .wpnc__main {
-	background-color: var( --color-white );
+	background-color: var( --studio-white );
```
"studio" has no meaning for people working on Calypso. I think it should be consistent with all other color prefixes in Calypso — `--color`.

porada replied:
I mentioned this in the notes:

> All app-specific color properties are now prefixed with `--color-` while generic colors are directly outsourced to Color Studio properties (prefixed with `--studio-`).

### Code duplication

I don’t think we can afford to rename all the generic colors, because all of that would end up in the output file, duplicating the amount of code that we really need.

```scss
:root {
   /* Defined directly in Color Studio (included automatically) */
  --studio-blue-50: #2271b1;
  --studio-blue-50-rgb: 34, 113, 177;

  /* Renamed later in color schemes (duplicated lines) */
  --color-blue-50: var( --studio-color-blue-50 );
  --color-blue-50-rgb: var( --studio-color-blue-50-rgb );
}
```

The above would be repeated for every single color in our palette. We used to rename the colors by relying on SCSS color variables instead, but those were phased out.

```scss
:root {
  --color-blue-50: #{$studio-color-blue-50};
  --color-blue-50-rgb: #{hex-to-rgb( $studio-color-blue-50 )};
}
```

The introduction of the color schemes marked the beginning of our migration away from SCSS color variables to CSS color properties. We used both formats inconsistently in our stylesheets (even for theming) when we only need one. Given that CSS custom properties are the future standard, there’s no question about which one we should use.

The `--studio` prefix isn’t ideal, but the color doc in Calypso will make it clear that we’re using Color Studio—which is also an open-source tool from Automattic (and which our other projects also rely on).

### Themable properties

If you take a look at the [default color scheme](https://github.com/Automattic/wp-calypso/blob/update/colors/packages/calypso-color-schemes/src/shared/color-schemes/_default.scss), you’ll notice that all themable properties are prefixed with `--color`. This means that all `--color` properties can be overwritten by another color scheme—and that’s the main difference from the `--studio` properties, which can’t be overwritten as they’re the building blocks of all the color schemes.

---

I’m open to other suggestions as to how to make that clearer, but the truth that you caught me somebody else’s code smell. When writing CSS, we shoud rely on `--color` properties wherever possible (`--color-surface` in this case).

ramonjd commented:
Just eye-balling signup cc @dwolfe @fditrapani @shaunandrews 

![New Project](https://user-images.githubusercontent.com/6458278/64212616-49aa9d80-ceed-11e9-872f-9ba77d4755b9.png)

Currently the signup primary color is `--color-primary` (#016087), which matches the current master bar color.

`--color-primary` is now `--studio-blue-50` (#2271b1) and the masterbar is `--studio-wordpress-blue-50` (#006088)

porada committed:
Rename color properties

porada commented:
@ramonjd Thanks for pointing that out. That’s my omission. The signup should use WordPress Blue for better continuity with our landing pages. Already pushed a fix:

<img width="940" alt="Screen Shot 2019-09-04 at 10 55 21" src="https://user-images.githubusercontent.com/2060580/64240628-80d77980-cf02-11e9-8ed1-ad635f84e0bb.png">

simison reviewed (commented):
Left a couple minor comments/questions.

Before I can test this you should rebase on top of `master` since this wasn't updated since August 20th and a lot of code has changed since. ;-) Pay attention e.g. to changes made in https://github.com/Automattic/wp-calypso/pull/35793

porada commented on client/components/notice/style.scss:
```diff
@@ -90,6 +97,22 @@
+  left: 15px;
+  width: 18px;
+  height: 18px;
+  border-radius: 9px;
```
Looks like Stylelint didn’t catch that. Fixed.

porada commented on client/components/accordion/docs/example.jsx:
```diff
@@ -65,12 +65,32 @@ AccordionExample.defaultProps = {
+				Etiam dictum odio elit, id faucibus urna elementum ac. Mauris in est nec tortor luctus
+				auctor ut a velit. Suspendisse vulputate lectus arcu, sed condimentum risus rutrum vitae.
+				Nullam sagittis ultricies nisl. Duis accumsan libero vel arcu sodales venenatis.
+			</Accordion>
```
They’re tied to this specific PR. Some component examples don’t showcase all possible cases, so I added those that would be impacted by the visual changes in this update.

porada commented on packages/calypso-color-schemes/package.json:
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@automattic/calypso-color-schemes",
-	"version": "1.0.0",
+	"version": "2.0.0",
```
Good idea. I’ll add that along with the color doc updates.

porada replied:
Resolving. Added to the checklist above.

porada commented on client/blocks/color-scheme-picker/style.scss:
```diff
@@ -1,38 +1,11 @@
 
-	&.is-classic-bright {
-		background-image: url( '/calypso/images/color-schemes/color-scheme-thumbnail-light.svg' );
+	@each $name in classic-blue classic-bright contrast nightfall midnight ocean sakura powder-snow sunset {
```
Good idea. ✅

porada commented:
@simison Thanks for the comments and the pointers! Will rebase that later today.

fditrapani commented:
The new Wordpress blue looks good to me for signup. Thanks @ramonjd and @porada!

porada commented:
@simison All rebased now.

I’ll still take a moment to systematize color properties in the codebase. I want to limit the number of cases where we rely on absolute colors (`--studio`) instead of theme colors (`--color`). Those changes won’t be visual as they’ll revolve primarily around grepping and renaming.

☝️ @drw158

porada committed:
Fix broken references

porada committed:
Tweak sidebar styles

sixhours commented:
I haven't had a chance to look at this in depth yet, but I noticed some weirdness when I first booted up the branch. This extra dark yellow dot appears on a warning notice when I view a stats page that has no stats. It doesn't happen on master, so I think it might be related to this PR:

<img width="187" alt="Screen Shot 2019-09-04 at 1 51 08 PM" src="https://user-images.githubusercontent.com/2124984/64279235-922d7f80-cf1c-11e9-99a2-f6307df80860.png">

Will dive in deeper soon. :)

sirreal commented on client/components/notice/index.jsx:
```diff
@@ -109,10 +124,14 @@ export class Notice extends Component {
 			<div className={ classes }>
 				<span className="notice__icon-wrapper">
-					<Gridicon className="notice__icon" icon={ icon || this.getIcon() } size={ 24 } />
+					{ iconNeedsDrop && <span class="notice__icon-wrapper-drop" /> }
```
There may be some positioning issues here. This addition is creating an additional circle [as noted by](https://github.com/Automattic/wp-calypso/pull/35949#issuecomment-528074972) @sixhours:

![Screen Shot 2019-09-05 at 10 07 23](https://user-images.githubusercontent.com/841763/64323784-6ca88080-cfc5-11e9-8b7f-08837af5384e.png)

porada replied:
@sixhours That’s on me. I’ll fix it right away. Thanks for reporting.

porada committed:
Tweak compact notice bars

porada committed:
Tweak compact notice bars

porada committed:
Use theme properties instead of direct color values

porada committed:
Colorize push notification illustration

porada commented:
@drw158 @simison @sixhours So I outsourced all color references to theme properties in places where the use of absolute color values/variables was unintentional. This means that in most cases, all color properties should start with `--color` instead of `--studio`.

There are a few cases where `--studio` colors (absolute colors) are used intentionally, which means they can’t be affected a change of scheme. Those are primarily Jetpack, WooCommerce, or very specific UI overrides.

Unless anybody finds a visual regression, all that’s left is to update the documentation. 🚀

drw158 commented:
This is really great. Thank for taking the time to use more semantic variables, like the surface and inverted text variables.

**High contrast scheme**
I think the intention with this scheme is to pass AAA on most UI elements. The updates decrease the contrast significantly on some things and probably only pass AA now. cc @sixhours 

left: changes / right: master
![Screen Shot 2019-09-05 at 10 26 15 AM](https://user-images.githubusercontent.com/618551/64356065-dcb80600-cfc7-11e9-84e0-ffa0489565d9.png)
![Screen Shot 2019-09-05 at 10 25 37 AM](https://user-images.githubusercontent.com/618551/64356066-dcb80600-cfc7-11e9-886b-3e1454e742bd.png)

Contrast thumbnail has a pink button

![Screen Shot 2019-09-05 at 10 39 50 AM](https://user-images.githubusercontent.com/618551/64357445-4507e700-cfca-11e9-9e55-be009bccee60.png)


**Ocean**

The new sidebar colors (left) don't look quite as nice. Maybe use the next lightest yellow shade for the selected item in the sidebar?

left: changes / right: master
![Screen Shot 2019-09-05 at 10 40 34 AM](https://user-images.githubusercontent.com/618551/64357185-d591f780-cfc9-11e9-939f-8e7d56031507.png)


**Sakura**

I think the current selected sidebar item colors (right) are nicer and go better with the overall feeling of the scheme.

left: changes / right: master
![Screen Shot 2019-09-05 at 10 38 58 AM](https://user-images.githubusercontent.com/618551/64357291-ffe3b500-cfc9-11e9-9163-b5b4a1fc8753.png)

**CC component**

What's up with the color on this component? Sometimes it's gray, and sometimes blue. In master it's green. It looks like the colors are applied in js maybe? I saw hex and rgb in the inspector.

left: changes / right: master
![Screen Shot 2019-09-05 at 10 37 51 AM](https://user-images.githubusercontent.com/618551/64357564-85676500-cfca-11e9-9943-aa7617501ed4.png)
![Screen Shot 2019-09-05 at 10 32 27 AM](https://user-images.githubusercontent.com/618551/64357565-85676500-cfca-11e9-8181-02b966264839.png)

drw158 commented:
Is there a plan to update Calypsofy pages?

porada commented:
@drw158 Thanks for detailed the review.

> I think the intention with this scheme is to pass AAA on most UI elements.

That wasn’t 100% clear to me. I’ll work on the compliance.

> Ocean
> Sakura

I intentionally updated the selected states to be more contrasting. The impact of those is better seen in grayscale.

> CC component

I updated this component to follow the Material approach and converted it to theme properties. That may explain why you encountered two color versions of the same component. Let me know if that wasn’t the case.

> Is there a plan to update Calypsofy pages?

What are these?

porada committed:
Improve accessibility under contrast color scheme

drw158 commented:
Calypsoify pages are the WP Admin pages that look like Calypso. The best example is the Plugins screen on Atomic sites. We use the Jetpack plugin to override WP Admin colors so that it looks like Calypso. Here's an example PR from when we did the first color update early this year: https://github.com/Automattic/jetpack/pull/11115

![Screen Shot 2019-09-05 at 1 25 10 PM](https://user-images.githubusercontent.com/618551/64369125-48f33380-cfe1-11e9-947c-da4c748887a3.png)

Currently schemes don't work on these pages, but we did update them to use the new 2019 colors. Also, Calypso colors don't apply at all in Gutenberg.

Other places to check and consider:

- Customizer (I thought we updated this, but it's still using the 2018 palette)
- Notifications (seen on the front end, not in Calypso)

![Screen Shot 2019-09-05 at 1 25 48 PM](https://user-images.githubusercontent.com/618551/64369130-4d1f5100-cfe1-11e9-9b07-2ce8a8af82c1.png)

drw158 commented:
"studio" probably doesn't mean anything to most people. Would it make more sense to use "brand" or just use "color"?

porada commented:
@drw158 Thanks for clarifying. As much as I’d like to, I won’t be able to include those at this time.

> Would it make more sense to use "brand" or just use "color"?

If this is about Automattic brands’ colors, then they’re already [defined here](https://github.com/Automattic/wp-calypso/blob/update/colors/packages/calypso-color-schemes/src/shared/color-schemes/_default.scss#L277-L282).

If that’s about the general use of color, I’d rather not alter the Color Studio build or redefine it. As [outlined here](https://github.com/Automattic/wp-calypso/pull/35949#discussion_r320669850), we should rely on theme variables first and foremost, and making the absolute color values easier to use encourages the anti-pattern.

> "studio" probably doesn't mean anything to most people

Having made the recent changes, I’m leaning towards the opinion that it’s completely fine. 🙂

---

There’s a case to be made that Automatticians are already aware that Color Studio is our internal color palette (#35793 is a good example). Other projects rely on Color Studio as a direct dependency, too.

And for those who’ll need to reference the absolute values, the updated documentation (which is coming later today) will outline the way to approach that.

☝️ @simison

simison commented:
Should there be a stylelint rule discouraging use of `--studio-` values? Eventually when they're "fixed" away we could make the rule to error, not just warn.

porada committed:
Tweak unread notification indicator

simison commented:
> Would you be willing to contribute at least a scaffolding for a rule like that? Given my inexperience with the custom linting rules and the fact that the GM is taking place next week, I’m afraid I can’t really put more on my plate if we want to ship those changes soon.

I'm equally inexperienced writing such rules, too. 😅  Oh I didn't mean that should be blocking this PR at all! Ship without such rule and have it in backlog to follow-up. ;-)

porada committed:
Update color documentation

porada committed:
Fix linting issues

porada commented:
Rebased once again and got rid of linting issues. Ready for the final review and hopefully to be merged. 🚀

simison commented on packages/calypso-color-schemes/README.md:
```diff
@@ -1,6 +1,8 @@
-This package contains a number of CSS custom properties and Sass variables used in Calypso.
+This package contains a number of CSS custom properties used in Calypso.
+
+![Color scheme thumbnails](screenshot@2x.png)
```
The image is nice but consider it _will_ get outdated. Is that an issue?

porada replied:
I don’t think it is. It represents the concept more than the actual themes and colors.

simison commented on client/components/notice/index.jsx:
```diff
@@ -109,10 +129,14 @@ export class Notice extends Component {
 			<div className={ classes }>
 				<span className="notice__icon-wrapper">
-					<Gridicon className="notice__icon" icon={ icon || this.getIcon() } size={ 24 } />
+					{ iconNeedsDrop && <span class="notice__icon-wrapper-drop" /> }
```
This `class=` is throwing `Warning: Invalid DOM property `class`.`. It should be `className`

porada replied:
Fixed in an upcoming push. Thanks!

simison reviewed (commented):
Just noting that I see this warning while testing but I'm not sure if it's coming from this PR:

<img width="735" alt="Screenshot 2019-09-09 at 15 28 19" src="https://user-images.githubusercontent.com/87168/64530702-79570c80-d316-11e9-9e98-bcded3404c8c.png">

simison reviewed (approved):
Tested different sections with Jetpack site and this seems to work. Obviously with almost 600 files I can't test everything. ;-)

Please rebase on top of `master` before merging to ensure you're not missing anything.

The only merge-blocking comment from is about `class`/`className`.

### Some notes:

3 Blue backgrounds in theme selection looks a bit funky, not sure why: 

<img width="1382" alt="Screenshot 2019-09-09 at 15 14 05" src="https://user-images.githubusercontent.com/87168/64530793-a4d9f700-d316-11e9-83a5-deab3763ae53.png">

Note also very non-contrasty "continue editing" (start editing in calypso editor and then close the editor before saving for that to appear).

Jetpack flow worked fine in my testing (use https://jurassic.ninja/ to get a test site):

<img width="1114" alt="Screenshot 2019-09-09 at 15 10 53" src="https://user-images.githubusercontent.com/87168/64530906-e10d5780-d316-11e9-89bc-c4b45cd7916e.png">
<img width="1127" alt="Screenshot 2019-09-09 at 15 10 35" src="https://user-images.githubusercontent.com/87168/64530907-e1a5ee00-d316-11e9-892b-ff6751dd822b.png">

In "uploaded themes" section three-dots menu didn't have enough contrast. Ok to fix in a follow-up PR:

<img width="315" alt="Screenshot 2019-09-09 at 15 04 15" src="https://user-images.githubusercontent.com/87168/64530949-00a48000-d317-11e9-839b-720952f6f9c0.png">

Sometimes these grey buttons have grey text color, sometimes little darker. I'm not sure what's the difference — button size?

<img width="1168" alt="Screenshot 2019-09-09 at 15 05 09" src="https://user-images.githubusercontent.com/87168/64531060-3e090d80-d317-11e9-9d88-f951f5a90274.png">
<img width="537" alt="image" src="https://user-images.githubusercontent.com/87168/64531347-f0d96b80-d317-11e9-9f7a-943874c9c411.png">


Icon at Settings->Performance for Jetpack site doesn't look great on Firefox:
<img width="743" alt="image" src="https://user-images.githubusercontent.com/87168/64531263-b66fce80-d317-11e9-90a0-ebc7cedfdd22.png">


Production for comparison — also broken but in different way:
<img width="757" alt="image" src="https://user-images.githubusercontent.com/87168/64531252-b1128400-d317-11e9-93c5-53dd6e43646d.png">

porada commented:
@simison Thanks for the review!

The sidebar has undergone a recent refactor, so that’s where that console warning might have come from. This update doesn’t touch any sidebar-related JSX bits.

> 3 Blue backgrounds in theme selection looks a bit funky, not sure why

I guess it’s because the part of of the customizer or the theme picker are based on legacy views. I’ve tried fine-tuning those areas, but quickly realized the scope size increased too quickly. That bit hasn’t been updated v1 colors even. I’ll create a separate issue for those.

> Sometimes these grey buttons have grey text color, sometimes little darker. I'm not sure what's the difference

That could be size and/or the selected color scheme.

simison commented:
> > Sometimes these grey buttons have grey text color, sometimes little darker. I'm not sure what's the difference
>
> That could be size and/or the selected color scheme.

Forgot to add that I'm testing with "default" color scheme.

>This is the same issue, but those SVG images define a different scaling method. I’ll fix that.

Prolly fine for another PR to keep this more focused. ;-)

drw158 commented:
Is compact buttons have gray text and it's not unique to this PR

drw158 commented:
Looks like the jetpack nux is using a blue color for the text when it should be gray

"Now that you're set up..."

porada commented:
Moved the bugs and missing colors updates mentioned above into separate issues. Rebased and merging soon.

Thanks once again for all the reports and the help on this update.

simison commented:
It was mentioned a few times outside Internets that browser inspector has gotten very slow after this PR was merged.

cc at least @ramonjd and I can't remember who else mentioned the same. :-) What was your browser?

Aurorum commented:
I also feel that btw :)

porada commented:
I experienced the very same problem while working on this, but I’m not sure yet if this update was the one that introduced it. We actually cut down on the number of custom properties with this PR.

simison commented:
I think the issue might be that all few hundred color variables are appended to _each_ CSS bundle loaded during Calypso session. That'll cause each new CSS bundle download overwriting previous variables in memory but I suppose all those thousands of variables kept in inspector's memory?

<img width="865" alt="Screenshot 2019-09-15 at 16 59 53" src="https://user-images.githubusercontent.com/87168/64927600-e49f5380-d7da-11e9-8589-a08c0af2dfd7.png">

Seems like color variables should be loaded just once initially (maybe even inline on page render?) and not on every bundle?

cc @sgomes as this is clearly a performance issue.

sgomes commented:
Thanks for the ping, @simison! This PR seems to have introduced some performance issues, both on the CSS and the JS side of things. I haven't had a chance to look at this thoroughly yet, as the way it was merged (with a merge commit) makes it very hard to isolate from other changes that got interspersed throughout.

Adding it to my backlog.

ramonjd commented:
Thanks @simison and @sgomes for looking into this.

I experimented a couple of times by resetting my local branch to the commit immediately previous to [2fcbd46c402793e5b7b17014627a15da105e01ee](https://github.com/Automattic/wp-calypso/commit/2fcbd46c402793e5b7b17014627a15da105e01ee), reinstalling node modules and rebooting Calypso, then resetting it to 2fcbd46c402793e5b7b17014627a15da105e01ee.

Confirmed that this effect is very noticeable in Chrome, Chromium and Firefox.

It's relatively tolerable if you switch away from the styles pane, that is, you can inspect elements and perform other important tasks in debugger, but inspecting and editing styles is no longer fun. :)

sirreal commented on packages/calypso-color-schemes/.npmignore:
```diff
@@ -1 +1,4 @@
+.eslintrc.js
+bin
+screenshot@2x.png
+src/__color-studio
```
This broke SASS in the published package. Fixed in #36171.

sirreal commented on packages/calypso-color-schemes/CHANGELOG.md:
```diff
@@ -0,0 +1,18 @@
+
+* Updated [Color Studio](https://color-studio.blog), the primary dependency, to the most recent version.
+* Following the [files](https://github.com/Automattic/color-studio/blob/master/dist/color-properties.css) [generated](https://github.com/Automattic/color-studio/blob/master/dist/color-properties-rgb.css) by Color Studio, updated all properties to use consistent names (including index numbers).
+* Phased out all SCSS variables in favor of CSS custom properties.
```
Why did this happen? This breaks dependent SASS pipelines. Importing CSS is very different from importing SASS in that it generates output.

I'd speculate that this is the cause of https://github.com/Automattic/wp-calypso/pull/35949#issuecomment-531599278 and is likely related to a bump in bundle sizes @sgomes has been tracking down.

porada replied:
As I already mentioned in one of the comments, it was a handful of the _remaining_ SCSS variables that were phased out. Those had relied on SCSS color functions like `darken` or `lighten`, and that was the reason why the original PR introducing `calypso-color-schemes` left them out.

In other words, I completed the switch from one SCSS variables to CSS custom properties that had already been done in 95%.

sirreal reviewed (commented):
This was a big change, thanks for taking that on 🙂 

Some issues have appeared in and around Calypso that seem to be related to this changeset that we'll need to work to track down.

If we find ways to keep changesets smaller, it should help with reviews in addition to making it easier to track down the source of issues when they are introduced.

sirreal commented:
I'm working on addressing the issues here in https://github.com/Automattic/wp-calypso/pull/36175

sgomes commented:
I'm seeing a few things in this PR that I'm having trouble grokking and understanding correctly.

This PR touches a lot of stuff, and it seems to make a bunch of different types of changes:
- It renames some variables across the application
- It changes how variables are loaded
- It changes how JS accesses the various variables
- It makes specific colour changes for a particular theme

Combined with the large number of files changed in this PR, this makes it really hard to reason about and to track down some negative effects it appears to have introduced.

This was compounded by the fact that the PR was merged with a merge commit, interspersing it with other unrelated commits, and the branch is now deleted. (This is no longer possible, as the repo will now only accept squash merges)

Is it possible to get a bit more clarity and background on what's happening here?
- Why was everything switched to CSS custom properties (which are great, and I love, by the way!) if we have a SCSS pipeline already, that functions better with SCSS variables?
- What are we doing on browsers that we still care about that don't CSS custom properties? I see lines like `background: rgba( var( --color-surface-rgb ), 0 );`. Does this get transformed at some point with some fallback for e.g. IE 11?
- If the goal is to switch to CSS custom properties, that are a runtime construct, why are we pre-generating all the themes and loading them all, instead of e.g. generating a CSS file for each theme that redefines the custom properties and thus affects the whole application?

In general, it would likely be best if we were to split large changes like this across multiple PRs that build on each other, and would let reviewers focus on one type of change at a time.

porada commented:
> * Why was everything switched to CSS custom properties?
> * What are we doing on browsers that we still care about that don't CSS custom properties?

This PR built on the existing concepts introduced earlier with Calypso color schemes. It didn’t really touch any stylesheets postprocessing.

There were a few SCSS variables left in places that relied on SCSS color functions like `darken` or `lighten`. Those have been refactored to use predefined colors.

porada commented:
> It changes how JS accesses the various variables

@sgomes Could you elaborate on that?

porada commented:
> In general, it would likely be best if we were to split large changes like this across multiple PRs that build on each other, and would let reviewers focus on one type of change at a time.

Noted. Although, please note some of the issues mentioned above had been introduced before I started working on this.

I have a local copy of this branch and can bring it back. What else can I do to help?
